Fixes 491 [FEAT] Allow disks to be hidden/archived

Berry-95 opened the pull request:
- Add archived to device type & db.
- Add archive/unarchive handlers to webapp backend.
- Add archive toggle & styling to webapp frontend.

saberesf committed:
491 [FEAT] Allow disks to be hidden/archived
- Add archived to device type & db
- Add archive/unarchive handlers to webapp backend
- Add archive toggle & styling to webapp frontend

Berry-95 commented:
Fixes https://github.com/AnalogJ/scrutiny/issues/491

Berry-95 commented:
![localhost_8080_web_dashboard](https://github.com/user-attachments/assets/10fcb0d6-1400-4a28-8cd6-6def3b407f1d)
![localhost_8080_web_dashboard (1)](https://github.com/user-attachments/assets/9826438d-4952-4e82-8d26-692468c284f6)

AnalogJ commented:
Interesting, this looks like a pretty great feature! 

Your changes look clean as well, thanks for the PR @Berry-95 ! 🥳

Berry-95 committed:
Merge branch 'master' into 491-FEAT-Allow-disks-to-be-archived

AnalogJ commented:
Hey @Berry-95  this looks 99% there. Looks like theres a type definition mis-match in the frontend thats broken, can you get that fixed?

Berry-95 commented:
Hey @AnalogJ I did notice the CI fail, I'll check it out when I have some time, hopefully soon.

saberesf committed:
491 [FEAT] Allow disks to be hidden/archived
- Fix mock device type definition mismatch in the frontend.
- Make DeviceModel archived field optional.

Berry-95 commented:
@AnalogJ Should be sorted now.
